packages/vendure-plugin-accept-blue/src/api/accept-blue-service.ts

```diff
@@ -68,31 +79,186 @@ export class AcceptBlueService {
+    );
+    if (!paymentMethod) {
+      throw new UserInputError(
+        'No Saved Credit Card with this toknized nonce code'
+      );
+    }
+    return await this.payWithCardPaymentMethodDetails(
       ctx,
       order,
       amount,
       client,
-      paymentMethod.id
+      paymentMethod.id,
+      ccDetails
     );
   }
 
+  async payWithCheck(
+    ctx: RequestContext,
+    order: Order,
+    amount: number,
+    client: AcceptBlueClient,
+    ccDetails: CheckPaymentInput
+  ): Promise<HandlePaymentResult> {
+    if (!order.customer) {
+      throw new UserInputError(`Order must have a customer`);
+    }
+    const customer = await client.getOrCreateCustomer(
+      order.customer.emailAddress
+    );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
+    // Save payment method
+    const paymentMethod = await client.getOrCreateCheckPaymentMethod(
+      customer.id,
+      ccDetails
+    );
+    return await this.payWithCheckPaymentMethodDetails(
+      ctx,
+      order,
+      amount,
+      client,
+      paymentMethod.id,
+      ccDetails
+    );
+  }
+
+  async payWithCardPaymentMethodDetails(
+    ctx: RequestContext,
+    order: Order,
+    amountDueNow: number,
+    client: AcceptBlueClient,
+    paymentMethodId: number,
+    ccDetails: CreditCardPaymentInput | TokenPaymentMethodInput
+  ): Promise<HandlePaymentResult> {
+    const recurringSchedules =
+      await this.createRecurringPaymentScheduleForOrder(
+        ctx,
+        order,
+        client,
+        paymentMethodId
+      );
+    let creditCardChargeInput:
+      | AcceptBlueCreditCardChargeTransactionInput
+      | AcceptBlueTokenizedCreditCardChargeTransactionInput;
+    if ((ccDetails as any).card) {
+      creditCardChargeInput = {
+        amount: amountDueNow,
+        card: (ccDetails as CreditCardPaymentInput).card,
+        expiry_month: ccDetails.expiry_month,
+        expiry_year: ccDetails.expiry_year,
+      };
+    } else if ((ccDetails as any).source) {
+      creditCardChargeInput = {
+        amount: amountDueNow,
+        source: (ccDetails as TokenPaymentMethodInput).source,
+        expiry_month: ccDetails.expiry_month,
+        expiry_year: ccDetails.expiry_year,
+      };
+    } else {
+      throw new Error('Not Implemented');
+    }
+    const chargeResult = await client.createCharge(creditCardChargeInput);
+    return {
+      customerId: `${order?.customer?.customFields.activeBlueCustomerId}`,
+      paymentMethodId,
+      recurringScheduleResult: recurringSchedules,
+      chargeResult,
+    };
+  }
+
+  async payWithCheckPaymentMethodDetails(
+    ctx: RequestContext,
+    order: Order,
+    amountDueNow: number,
+    client: AcceptBlueClient,
+    paymentMethodId: number,
+    ccDetails: CheckPaymentInput
+  ): Promise<HandlePaymentResult> {
```

Comment:
How come we accept the `ccDetails` AND `paymentMethodId`? Can't we follow this flow for all payment types (check, card and tokenized)?

Get or Create customer > get Or Create Payment method for customer > **Create payment with saved payment method** (not passing around the details)

Reply:
We use the `paymentMethodId` to create recurring schedules and we use `ccDetails` to create a one time charge. What do you think about this approach @martijnvdbrug ?

Reply:
@dalyathan Thats fine for now! I was just hoping we could do a one time charge with saved payment methods, so that customers could in the future check out with just a saved payment method.